tasks/eth/011-proto-ver+ownership-transfers/SignFromJson.s.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson as OriginalSignFromJson} from "script/SignFromJson.s.sol";
+import {Proxy} from "@eth-optimism-bedrock/src/universal/Proxy.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {ProtocolVersions, ProtocolVersion} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "mainnet";
+    string constant l2ChainName = "op";
+
+    uint256 constant protoVerEcotone = 0x0000000000000000000000000000000000000006000000000000000000000000;
+    uint256 constant protoVerFjord = 0x0000000000000000000000000000000000000007000000000000000000000000;
+
+    // Safe contract for this task.
+    GnosisSafe foundationUpgradesSafe = GnosisSafe(payable(vm.envAddress("FOUNDATION_SAFE")));
+    GnosisSafe foundationOperationsSafe = GnosisSafe(payable(vm.envAddress("FOUNDATION_OP_SAFE")));
+
+    // All L1 proxy addresses.
+    Types.ContractSet proxies;
+
+    function setUp() public {
+        proxies = _getContractSet();
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view override {
+        address[] memory allowed = new address[](3);
+        allowed[0] = address(foundationOperationsSafe);
+        allowed[1] = proxies.ProtocolVersions;
+        allowed[2] = proxies.SystemConfig;
+        super.checkStateDiff(accountAccesses, allowed);
+
+        checkProtocolVersions();
+        checkSystemConfig();
```

Comment:
nit, these methods are typically called in `_postCheck`. The way we think about it is: one of the post-checks is the state diff assertions from `checkStateDiff`, other post-checks are e.g. `checkSystemConfig`